assets/src/components/detours/diversionPage.tsx

```diff
@@ -1,7 +1,107 @@
-import React from "react"
+import React, { useEffect, useMemo, useState } from "react"
 import { DiversionPanel, DiversionPanelProps } from "./diversionPanel"
 import { DetourMap } from "./detourMap"
-import { Shape } from "../../schedule"
+import { Shape, ShapePoint } from "../../schedule"
+import { LatLngLiteral } from "leaflet"
+import { fetchDetourDirections } from "../../api"
+import {
+  latLngLiteralToShapePoint,
+  shapePointToLatLngLiteral,
+} from "../../util/pointLiterals"
+
+const useDetourDirections = (shapePoints: ShapePoint[]) => {
+  const [detourShape, setDetourShape] = useState<LatLngLiteral[]>([])
+
+  useEffect(() => {
+    let shouldUpdate = true
+
+    fetchDetourDirections(shapePoints).then((detourShape) => {
+      if (detourShape && shouldUpdate) {
+        setDetourShape(detourShape.coordinates.map(shapePointToLatLngLiteral))
+      }
+    })
+
+    return () => {
+      shouldUpdate = false
+    }
+  }, [shapePoints])
```

Comment:
I don't think this is memoizing properly. When I run this, `fetchDetourDirections` is being called potentially infinitely many times (well, 40 per minute before the API 500's)

Reply:
I think [this](https://github.com/mbta/skate/pull/2364/files#r1456500283) fixes it